src/status_im2/subs/shell.cljs

```diff
@@ -17,44 +19,55 @@
                       (first images)))})))
 
 (defn get-card-content
-  [chat communities]
-  (let [last-message (:last-message chat)]
+  [chat communities group-chat primary-name]
+  (let [{:keys [content-type content deleted? outgoing deleted-for-me?] :as last-message}
+        (:last-message chat)]
     (merge
      (when last-message
-       (case (:content-type last-message)
-         (constants/content-type-text
-          constants/content-type-emoji)
+       (cond
+         (or deleted-for-me? deleted?)
          {:content-type constants/content-type-text
-          :data         (get last-message :content)}
+          :data         {:text
+                         (if (or deleted-for-me? outgoing)
+                           (i18n/label :t/you-deleted-a-message)
+                           (if group-chat
+                             (i18n/label :t/user-deleted-a-message {:user primary-name})
+                             (i18n/label :t/this-message-was-deleted)))}}
+
+         (#{constants/content-type-text
+            constants/content-type-emoji}
+          content-type)
+         {:content-type constants/content-type-text
+          :data         content}
 
          ;; Currently mock image is used as placeholder,
          ;; as last-message don't have image
          ;; https://github.com/status-im/status-mobile/issues/14625
-         constants/content-type-image
+         (= content-type constants/content-type-image)
          {:content-type constants/content-type-image
           :data         [{:source (resources/get-mock-image :photo2)}]}
 
          ;; Same for sticker, mock image is used
-         constants/content-type-sticker
+         (= content-type constants/content-type-sticker)
          {:content-type constants/content-type-sticker
           :data         {:source (resources/get-mock-image :sticker)}}
 
          ;; Mock Image
-         constants/content-type-gif
+         (= content-type constants/content-type-gif)
          {:content-type constants/content-type-gif
           :data         {:source (resources/get-mock-image :gif)}}
 
-         constants/content-type-audio
+         (= content-type constants/content-type-audio)
          {:content-type constants/content-type-audio
           :data         (datetime/ms-to-duration (:audio-duration-ms last-message))}
 
-         constants/content-type-community
+         (= content-type constants/content-type-community)
          (let [community (get communities (:community-id last-message))]
            {:content-type constants/content-type-community
             :data         {:avatar         (community-avatar community)
                            :community-name (:name community)}})
 
-         nil))
+         :else nil))
```

Comment:
We don't need a default/fallback card?

Reply:
subscription only returns content for the card, and default case nil was just added for the case because it would have thrown an error if a message with an unimplemented content type was received. It's not a problem with cond, removed it.